odoo/addons/base/models/ir_qweb_fields.py

```diff
@@ -609,17 +609,25 @@ def get_available_options(self):
             width=dict(type='integer', string=_('Width'), default_value=600),
             height=dict(type='integer', string=_('Height'), default_value=100),
             humanreadable=dict(type='integer', string=_('Human Readable'), default_value=0),
+            quiet=dict(type='integer', string='Quiet', default_value=1),
+            mask=dict(type='string', string='Mask', default_value='')
         )
         return options
 
     @api.model
     def value_to_html(self, value, options=None):
+        if not value:
+            return ''
         barcode_symbology = options.get('symbology', 'Code128')
         barcode = self.env['ir.actions.report'].barcode(
             barcode_symbology,
             value,
-            **{key: value for key, value in options.items() if key in ['width', 'height', 'humanreadable']})
-        return u'<img src="data:png;base64,%s">' % base64.b64encode(barcode).decode('ascii')
+            **{key: value for key, value in options.items() if key in ['width', 'height', 'humanreadable', 'quiet', 'mask']})
+        img_attributes = {k[4:]: v for k, v in options.items() if k.startswith('img_')}
+        if 'alt' not in img_attributes:
+            img_attributes.update({'alt': 'Barcode %s' % value})
+        attributes = ' '.join(['%s="%s"' % attrs for attrs in img_attributes.items()])
+        return u'<img src="data:png;base64,%s" %s>' % (base64.b64encode(barcode).decode('ascii'), attributes)
```

Comment:
That's not your code, but for some reason this has been slightly wrong since day 1: the first parameter after `data:` is a mimetype and should be `image/png`. Fortunately, browsers do all sorts of mimetype sniffing ;-)

nits: if you're touching the line, you can also remove the useless `u''` prefix. The parameter for `decode()` is useless too (12.0+ = P3 only)

```suggestion
        return '<img src="data:image/png;base64,%s" %s>' % (base64.b64encode(barcode).decode(), attributes)
```